app/controllers/work_packages/progress_controller.rb

```diff
@@ -111,28 +120,29 @@ def modal_class
     end
   end
 
-  def set_work_package
+  def find_work_package
     @work_package = WorkPackage.visible.find(params[:work_package_id])
-  rescue ActiveRecord::RecordNotFound
-    @work_package = WorkPackage.new
+  end
+
+  def make_fake_initial_work_package
+    initial_params = params["work_package"]["initial"]
+      .slice(*%w[estimated_hours remaining_hours done_ratio status_id])
+      .permit!
+    @work_package = WorkPackage.new(initial_params)
+    @work_package.clear_changes_information
```

Comment:
Is this so it doesn't pollute the `model.<field_name>_changed?` calls?

Reply:
Yes exactly, so it acts like the work package existed with the initial values, like when it's loaded from db.

```ruby
>> wp = WorkPackage.new(done_ratio: 14)
=>
#<WorkPackage:0x000000031a7f6698
...
>> wp.changes
=> {"done_ratio"=>[nil, 14]}
>> wp.clear_changes_information
=> nil
>> wp.changes
=> {}
```